packages/components/src/components/primitives/Link/Link.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { FC } from 'react';
+import {
+  Box,
+  Link as ChakraLink,
+  Text,
+  Flex,
+  FlexboxProps,
+} from '@chakra-ui/react';
+
+import { LinkProps } from './types';
+
+export const Link: FC<LinkProps> = ({
+  children,
+  icon,
+  iconPosition = 'trailing',
+  size = 'm',
+  ...props
+}) => {
+  const Icon: React.FC = () =>
+    React.cloneElement(icon, {
+      size,
+    });
+
+  const flexDirection = {
+    trailing: 'row',
+    leading: 'row-reverse',
+  }[iconPosition] as FlexboxProps['flexDirection'];
```

Comment:
I think we should do the opposite. Ideally `leading` should behave like a regular row and `trailing` like row reverse.